l10n_eu_nace/models/res_partner_nace.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.osv.expression import AND
+
+
+class ResPartnerNace(models.Model):
+
+    _name = 'res.partner.nace'
+    _description = 'European NACE partner category'
+    _order = 'code'
+    _parent_store = True
+
+    name = fields.Char(index=True, translate=True)
+    parent_id = fields.Many2one(comodel_name="res.partner.nace", index=True)
+    code = fields.Char(index=True)
+    child_ids = fields.One2many(
+        'res.partner.nace', 'parent_id', string='NACE subcategories'
+    )
+    active = fields.Boolean(index=True, default=True)
+    parent_path = fields.Char(index=True)
+    partner_ids = fields.One2many(
+        comodel_name="res.partner", inverse_name="nace_id", string="Partners"
+    )
+
+    @api.multi
+    def name_get(self):
+        res = []
+        for category in self:
+            if self._context.get('nace_display') == 'short':
+                res.append(
+                    (category.id, "[%s] %s" % (category.code, category.name))
+                )
+            else:
+                names = []
+                current = category
+                while current:
+                    if current.code:
+                        names.append("[%s] %s" % (current.code, current.name))
+                    else:
+                        names.append(current.name)
+                    current = current.parent_id
+                res.append((category.id, ' / '.join(reversed(names))))
+        return res
+
+    @api.model
+    def name_search(self, name, args=None, operator='ilike', limit=100):
+        args = list(args or [])
+        domain = AND([[("code", operator, name)], args])
+        if not (name == "" and operator == "ilike"):
+            domain = ["|"] + domain
+        return super().name_search(name, domain, operator, limit)
```

Comment:
... this:

```suggestion
        return super().name_search(name or self._rec_name, domain, operator, limit)
```

And define a `_rec_name` in the model (there's none AFAICS).

Reply:
I don't get you, 

argument name contain the search value, in case it's empty it will search for nace with name "display_name"?

IMO, there is no way to override name_search without changing the domain in unsafe way due to _name_search implementation in base model.

Avoiding the security leak has no added value as the leak is in base model

This explain why many of OCA and Odoo modules when overriding name_search don't send back to super and return the result of normal search and for the same reasons you last fix suggestions broke the tests.

So, I decide to give up on name_search and use a computed with a dedicated search method